ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py

```diff
@@ -715,3 +715,82 @@ def test_torch_svdvals(
         on_device=on_device,
         A=x,
     )
+
+
+# tensorinv
+@st.composite
+def _tensorinv_helper(draw):
+    def factors(x):
+        result = [1, ]
+        i = 2
+        while i * i <= x:
+            if x % i == 0:
+                result.append(i)
+                if x // i != i:
+                    result.append(x // i)
+            i += 1
+        result.append(x)
+        return np.array(result)
+    ind = draw(helpers.ints(min_value=1, max_value=5))
+    product_half = draw(helpers.ints(min_value=2,max_value=25))
+    factors_list = factors(product_half)
+    shape = ()
+    while len(shape) < ind and ind > 2:
+        while np.prod(shape) < product_half:
+            a = factors_list[np.random.randint(len(factors_list))]
+            shape += (a,)
+        if np.prod(shape) > product_half:
+            shape = ()
+        while len(shape) < ind:
+            shape += (1,)
+        if np.prod(shape) == product_half and len(shape) == ind:
+            shape += shape[::-1]
+            break
+    if ind == 1:
+        shape += (product_half, product_half)
+    if ind == 2:
+        shape += (1, product_half, product_half, 1)
+    shape_cor = ()
+    for i in shape:
+        shape_cor += (int(i),)
+    dtype = draw(helpers.get_dtypes("float"))
+    input = draw(helpers.array_values(
```

Comment:
Why not use the usual `dtype_and_values`  that allows to control the dimensions? You can pass `shape=shape_cor` and `available_dtypes=dtype`.

Reply:
I tried that one first, it didn't draw shapes I passed when I set `min_dim=2`. It added one extra dimension. Without the parameter It threw HealtyCheck error saying that size is too large. Working on it, will have time at weekend.

Reply:
update:
I wanted to test my shape generation part if it works well. I assumed `np.linalg.tensorinv` will work.  
```py
def test():
    def factors(x):
        result = [1, ]
        i = 2
        while i * i <= x:
            if x % i == 0:
                result.append(i)
                if x // i != i:
                    result.append(x // i)
            i += 1
        result.append(x)
        return np.array(result)

    ind = np.random.randint(2, 6)
    product_half = np.random.randint(2, 25)
    factors_list = factors(product_half)
    shape = ()
    while len(shape) < ind and ind > 2:
        while np.prod(shape) < product_half:
            a = factors_list[np.random.randint(len(factors_list))]
            shape += (a,)
        if np.prod(shape) > product_half or len(shape) > ind:
            shape = ()  
        while len(shape) < ind and shape != ():
            shape += (1,)
        if np.prod(shape) == product_half:
            shape += shape[::-1]
            break
    if ind == 1 and shape == ():
        shape += (product_half, product_half)
    if ind == 2 and shape == ():
        shape += (1, product_half, product_half, 1)
    shape_cor = ()
    for i in shape:
        shape_cor += (int(i),)
    return shape_cor, ind
x, y = test()
for i in range(100000):
    try:
        np.linalg.tensorinv(np.random.rand(*x), ind=y)

    except:
        print('failed')
```

I found that generated shape is not satisfying the product condition in some rare cases. So, I fixed that and tested if any of the generated `shape` and `ind` pairs are invalid for tensorinv function. By using `np.random.rand`, invertibility condition is satisfied without any extra code.

Then I used both helpers `dtype_and_values` and `array_values`. I went through step by step. As you can see from the code on above, invertibility the only condition left to satisfy. I used `ivy.det` bigger than epsilon as the filter and got healthy check error that saying generated data is so big. This make sense because hypothesis helpers draws array values and since we pass some relatively big shapes to the helpers.

Reply:
I forgot to mention checking determinant with `ivy.linalg.det` or `np.linalg.det` didn't work.
```py
shape_cor, ind = ((1, 2, 2, 1), 2)
x = np.random.rand(*shape_cor)
np.linalg.det(x)                             # first
np.linalg.tensorinv(x, ind=ind)     # second
```
First one throws an `LinAlgError: Last 2 dimensions of the array must be square`  second one works fine. Healthy check error comes when I use this as a filter `filter(lambda x: np.linalg.cond(x[1]) < 1 / sys.float_info.epsilon)`

Reply:
Thanks for the update! You're almost done I think 🙂